tests/pytests/statistics_checker.py

```diff
@@ -0,0 +1,308 @@
+        self.json_stat = json_statistics
+        self.categories = categories
+        self.backends_combination = backends_combination
+        self.start_time = time_period[0]
+        self.end_time = time_period[1]
+        self.real_backends = routes.get_address_backends(routes.addresses()[0])
+        self.backends_groups = {}
+        for i in self.real_backends:
+            self.backends_groups[i] = routes.get_address_backend_group(routes.addresses()[0], i)
+
+    @staticmethod
+    def convert_timestamp(value):
+        return datetime.fromtimestamp(value['tv_sec']) + timedelta(microseconds=value['tv_usec'])
+
+    @staticmethod
+    def __check_timestamp(value):
+        '''checks timestamp values'''
+        assert value['tv_sec'] >= 0
+        assert value['tv_usec'] >= 0
+
+    def __check_global_stat(self):
+        '''checks top-level values of statistics which is common for all categories'''
+        assert self.json_stat['monitor_status'] == 'enabled'
+        MonitorStatsChecker.__check_timestamp(self.json_stat['timestamp'])
+        assert self.json_stat['string_timestamp']
+        statistics_timestamp = MonitorStatsChecker.convert_timestamp(self.json_stat['timestamp'])
+        assert self.start_time < statistics_timestamp < self.end_time
+
+    def check_json_stat(self):
+        '''
+        checks common statistics for all categories.
+        Determines which categories should be presented and run corresponding checkers
+        '''
+        assert self.json_stat
+        self.__check_global_stat()
+
+        for category, checker in self.checkers.items():
+            if self.categories & category:
+                checker()
+
+    def __check_backend_json(self, backend_json, backend_id):
+        '''checks common part of one backend json'''
+        assert backend_json['backend_id'] == backend_id
+        assert backend_json['status']['state'] == 1
+        assert backend_json['status']['string_state']
+        assert backend_json['status']['defrag_state'] == 0
+        assert backend_json['status']['string_defrag_state']
+        MonitorStatsChecker.__check_timestamp(backend_json['status']['last_start'])
+        assert backend_json['status']['string_last_time']
+        last_start = MonitorStatsChecker.convert_timestamp(backend_json['status']['last_start'])
+        assert last_start < self.end_time
+        assert backend_json['status']['last_start_err'] == 0
+        assert backend_json['status']['read_only'] == False
+
+    def __check_backends_common(self):
+        '''checks common part of backends and checks that all requested backends are presented in json'''
+        if self.backends_combination is None:
+            self.backends_combination = self.real_backends # if backends_combination is None
+                                                           # statistics requested for all existing backends
+        assert len(set(self.json_stat['backends'].keys()).difference(self.backends_combination))
```

Comment:
What does this assert check?

Reply:
It check that answer contain all backends which was requested. I use len() because the set has no empty() func.